### PR TITLE
Add script to output in DIMACS format.

### PR DESCRIPTION
**Pull request by johntyree**
This script outputs our scenarios in the standard DIMACS format that many other sat solvers can read.

If you have the C++ implementation of minisat available, it can also feed it through there to see what kind of solution it provides.

### CODE REVIEW

**Comment by johntyree** [comment]
Hmm... Not sure what these millions of other ugly commits are... rebasing.

**Comment by johntyree** [comment]
Example output:
```
$ python scripts/scenario_to_dimacs.py simplesat/tests/simple_numpy.yaml 
c KEY
c {"1": "MKL 10.2-1", "2": "MKL 10.3-1", "3": "numpy 1.7.1-1", "4": "numpy 1.8.1-1"}
p cnf 4 5
-3 2 0
-4 -3 0
-2 -1 0
-4 2 0
3 4 0
```

File: scripts/scenario_to_dimacs.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    return Transaction(pool, solution_ids, installed_map)
+
+
+def main(argv=None):
+    argv = argv or sys.argv[1:]
+
+    p = argparse.ArgumentParser()
+    p.add_argument("scenario", help="Path to the YAML scenario file.")
+    p.add_argument("--solve", action="store_true")
```
Comment (cournape) [inline scripts/scenario_to_dimacs.py]: can we have help for those 3 options ?

**Comment by cournape** [comment]
One comment regarding help, then LGTM

**Pull request merged**